lib/aadutils.js

```diff
@@ -155,3 +156,21 @@ exports.rsaPublicKeyPem = (modulusB64, exponentB64) => {
 
   return pem;
 };
+
+// used for c_hash and at_hash validation
+// case (1): content = access_token, hashProvided = at_hash
+// case (2): content = code, hashProvided = c_hash
+exports.checkHashValueRS256 = (content, hashProvided) => {
+  // step 1. hash the content
+  var digest = crypto.createHash('sha256').update(content).digest();
```

Comment:
The default encoding in update method is "utf8". We should hash the 'ascii' encoding of the input (code or access token).